### ISSUE TITLE
after generating new scaffolding rails complains about undefined method `search' 

### ISSUE THREAD

**Issue by ozpozner** [rivsc/Beautiful-Scaffold#10]
After fighting with the asstes bug I have encountered:
undefined method `search' for #Class:0x321a950 
****\* around line 15 *****
    @q = Location.search(
      params[:q]
    )

---

NoMethodError - undefined method `search' for #<Class:0x321a950>:
  activerecord (4.1.0) lib/active_record/dynamic_matchers.rb:26:in`method_missing'
  app/controllers/locations_controller.rb:15:in `index'
  actionpack (4.1.0) lib/action_controller/metal/implicit_render.rb:4:in`send_action'
  actionpack (4.1.0) lib/abstract_controller/base.rb:189:in `process_action'
  actionpack (4.1.0) lib/action_controller/metal/rendering.rb:10:in`process_action'
  actionpack (4.1.0) lib/abstract_controller/callbacks.rb:20:in `block in process_action'
  activesupport (4.1.0) lib/active_support/callbacks.rb:113:in`call'
  activesupport (4.1.0) lib/active_support/callbacks.rb:149:in `block in halting_and_conditional'
  activesupport (4.1.0) lib/active_support/callbacks.rb:229:in`block in halting'
  activesupport (4.1.0) lib/active_support/callbacks.rb:229:in `block in halting'
  activesupport (4.1.0) lib/active_support/callbacks.rb:166:in`block in halting'
  activesupport (4.1.0) lib/active_support/callbacks.rb:166:in `block in halting'
  activesupport (4.1.0) lib/active_support/callbacks.rb:166:in`block in halting'
  activesupport (4.1.0) lib/active_support/callbacks.rb:86:in `run_callbacks'
  actionpack (4.1.0) lib/abstract_controller/callbacks.rb:19:in`process_action'
  actionpack (4.1.0) lib/action_controller/metal/rescue.rb:29:in `process_action'
  actionpack (4.1.0) lib/action_controller/metal/instrumentation.rb:31:in`block in process_action'
  activesupport (4.1.0) lib/active_support/notifications.rb:159:in `block in instrument'
  activesupport (4.1.0) lib/active_support/notifications/instrumenter.rb:20:in`instrument'
  activesupport (4.1.0) lib/active_support/notifications.rb:159:in `instrument'
  actionpack (4.1.0) lib/action_controller/metal/instrumentation.rb:30:in`process_action'
  actionpack (4.1.0) lib/action_controller/metal/params_wrapper.rb:250:in `process_action'
  activerecord (4.1.0) lib/active_record/railties/controller_runtime.rb:18:in`process_action'
  actionpack (4.1.0) lib/abstract_controller/base.rb:136:in `process'
  actionview (4.1.0) lib/action_view/rendering.rb:30:in`process'
  actionpack (4.1.0) lib/action_controller/metal.rb:195:in `dispatch'
  actionpack (4.1.0) lib/action_controller/metal/rack_delegation.rb:13:in`dispatch'
  actionpack (4.1.0) lib/action_controller/metal.rb:231:in `block in action'
  actionpack (4.1.0) lib/action_dispatch/routing/route_set.rb:80:in`dispatch'
  actionpack (4.1.0) lib/action_dispatch/routing/route_set.rb:48:in `call'
  actionpack (4.1.0) lib/action_dispatch/journey/router.rb:71:in`block in call'
  actionpack (4.1.0) lib/action_dispatch/journey/router.rb:59:in `call'
  actionpack (4.1.0) lib/action_dispatch/routing/route_set.rb:676:in`call'
  rack (1.5.2) lib/rack/etag.rb:23:in `call'
  rack (1.5.2) lib/rack/conditionalget.rb:25:in`call'
  rack (1.5.2) lib/rack/head.rb:11:in `call'
  actionpack (4.1.0) lib/action_dispatch/middleware/params_parser.rb:27:in`call'
  actionpack (4.1.0) lib/action_dispatch/middleware/flash.rb:254:in `call'
  rack (1.5.2) lib/rack/session/abstract/id.rb:225:in`context'
  rack (1.5.2) lib/rack/session/abstract/id.rb:220:in `call'
  actionpack (4.1.0) lib/action_dispatch/middleware/cookies.rb:560:in`call'
  activerecord (4.1.0) lib/active_record/query_cache.rb:36:in `call'
  activerecord (4.1.0) lib/active_record/connection_adapters/abstract/connection_pool.rb:621:in`call'
  activerecord (4.1.0) lib/active_record/migration.rb:380:in `call'
  actionpack (4.1.0) lib/action_dispatch/middleware/callbacks.rb:29:in`block in call'
  activesupport (4.1.0) lib/active_support/callbacks.rb:82:in `run_callbacks'
  actionpack (4.1.0) lib/action_dispatch/middleware/callbacks.rb:27:in`call'
  actionpack (4.1.0) lib/action_dispatch/middleware/reloader.rb:73:in `call'
  actionpack (4.1.0) lib/action_dispatch/middleware/remote_ip.rb:76:in`call'
  better_errors (1.1.0) lib/better_errors/middleware.rb:84:in `protected_app_call'
  better_errors (1.1.0) lib/better_errors/middleware.rb:79:in`better_errors_call'
  better_errors (1.1.0) lib/better_errors/middleware.rb:56:in `call'
  actionpack (4.1.0) lib/action_dispatch/middleware/debug_exceptions.rb:17:in`call'
  actionpack (4.1.0) lib/action_dispatch/middleware/show_exceptions.rb:30:in `call'
  railties (4.1.0) lib/rails/rack/logger.rb:38:in`call_app'
  railties (4.1.0) lib/rails/rack/logger.rb:20:in `block in call'
  activesupport (4.1.0) lib/active_support/tagged_logging.rb:68:in`block in tagged'
  activesupport (4.1.0) lib/active_support/tagged_logging.rb:26:in `tagged'
  activesupport (4.1.0) lib/active_support/tagged_logging.rb:68:in`tagged'
  railties (4.1.0) lib/rails/rack/logger.rb:20:in `call'
  actionpack (4.1.0) lib/action_dispatch/middleware/request_id.rb:21:in`call'
  rack (1.5.2) lib/rack/methodoverride.rb:21:in `call'
  rack (1.5.2) lib/rack/runtime.rb:17:in`call'
  activesupport (4.1.0) lib/active_support/cache/strategy/local_cache_middleware.rb:26:in `call'
  rack (1.5.2) lib/rack/lock.rb:17:in`call'
  actionpack (4.1.0) lib/action_dispatch/middleware/static.rb:64:in `call'
  rack (1.5.2) lib/rack/sendfile.rb:112:in`call'
  railties (4.1.0) lib/rails/engine.rb:514:in `call'
  railties (4.1.0) lib/rails/application.rb:144:in`call'
  rack (1.5.2) lib/rack/content_length.rb:14:in `call'
  thin (1.6.2) lib/thin/connection.rb:86:in`block in pre_process'
  thin (1.6.2) lib/thin/connection.rb:84:in `pre_process'
  thin (1.6.2) lib/thin/connection.rb:53:in`process'
  thin (1.6.2) lib/thin/connection.rb:39:in `receive_data'
  eventmachine-1.0.3-x86 (mingw32) lib/eventmachine.rb:187:in`run'
  thin (1.6.2) lib/thin/backends/base.rb:73:in `start'
  thin (1.6.2) lib/thin/server.rb:162:in`start'
  rack (1.5.2) lib/rack/handler/thin.rb:16:in `run'
  rack (1.5.2) lib/rack/server.rb:264:in`start'
  railties (4.1.0) lib/rails/commands/server.rb:69:in `start'
  railties (4.1.0) lib/rails/commands/commands_tasks.rb:81:in`block in server'
  railties (4.1.0) lib/rails/commands/commands_tasks.rb:76:in `server'
  railties (4.1.0) lib/rails/commands/commands_tasks.rb:40:in`run_command!'
  railties (4.1.0) lib/rails/commands.rb:17:in `<top (required)>'
  script/rails:6:in`<main>'
  script/rails:0:in `<main>'
******\* the generate controller is****
# encoding : utf-8

class LocationsController < BeautifulController

  before_filter :load_location, :only => [:show, :edit, :update, :destroy]

  # Uncomment for check abilities with CanCan
  #authorize_resource

  def index
    session[:fields] ||= {}
    session[:fields][:location] ||= (Location.columns.map(&:name) - ["id"])[0..4]
    do_select_fields(:location)
    do_sort_and_paginate(:location)

```
@q = Location.search(
  params[:q]
)

@location_scope = @q.result(
  :distinct => true
).sorting(
  params[:sorting]
)

@location_scope_for_scope = @location_scope.dup

unless params[:scope].blank?
  @location_scope = @location_scope.send(params[:scope])
end

@locations = @location_scope.paginate(
  :page => params[:page],
  :per_page => 20
).all

respond_to do |format|
  format.html{
    if request.headers['X-PJAX']
      render :layout => false
    else
      render
    end
  }
  format.json{
    render :json => @location_scope.all 
  }
  format.csv{
    require 'csv'
    csvstr = CSV.generate do |csv|
      csv << Location.attribute_names
      @location_scope.all.each{ |o|
        csv << Location.attribute_names.map{ |a| o[a] }
      }
    end 
    render :text => csvstr
  }
  format.xml{ 
    render :xml => @location_scope.all 
  }             
  format.pdf{
    pdfcontent = PdfReport.new.to_pdf(Location,@location_scope)
    send_data pdfcontent
  }
end
```

  end

  def show
    respond_to do |format|
      format.html{
        if request.headers['X-PJAX']
          render :layout => false
        else
          render
        end
      }
      format.json { render :json => @location }
    end
  end

  def new
    @location = Location.new

```
respond_to do |format|
  format.html{
    if request.headers['X-PJAX']
      render :layout => false
    else
      render
    end
  }
  format.json { render :json => @location }
end
```

  end

  def edit

  end

  def create
    @location = Location.create(params[:location])

```
respond_to do |format|
  if @location.save
    format.html {
      if params[:mass_inserting] then
        redirect_to locations_path(:mass_inserting => true)
      else
        redirect_to location_path(@location), :notice => t(:create_success, :model => "location")
      end
    }
    format.json { render :json => @location, :status => :created, :location => @location }
  else
    format.html {
      if params[:mass_inserting] then
        redirect_to locations_path(:mass_inserting => true), :error => t(:error, "Error")
      else
        render :action => "new"
      end
    }
    format.json { render :json => @location.errors, :status => :unprocessable_entity }
  end
end
```

  end

  def update

```
respond_to do |format|
  if @location.update_attributes(params[:location])
    format.html { redirect_to location_path(@location), :notice => t(:update_success, :model => "location") }
    format.json { head :ok }
  else
    format.html { render :action => "edit" }
    format.json { render :json => @location.errors, :status => :unprocessable_entity }
  end
end
```

  end

  def destroy
    @location.destroy

```
respond_to do |format|
  format.html { redirect_to locations_url }
  format.json { head :ok }
end
```

  end

  def batch
    attr_or_method, value = params[:actionprocess].split(".")

```
@locations = []    

Location.transaction do
  if params[:checkallelt] == "all" then
    # Selected with filter and search
    do_sort_and_paginate(:location)

    @locations = Location.search(
      params[:q]
    ).result(
      :distinct => true
    )
  else
    # Selected elements
    @locations = Location.find(params[:ids].to_a)
  end

  @locations.each{ |location|
    if not Location.columns_hash[attr_or_method].nil? and
           Location.columns_hash[attr_or_method].type == :boolean then
     location.update_attribute(attr_or_method, boolean(value))
     location.save
    else
      case attr_or_method
      # Set here your own batch processing
      # location.save
      when "destroy" then
        location.destroy
      end
    end
  }
end

redirect_to :back
```

  end

  def treeview

  end

  def treeview_update
    modelclass = Location
    foreignkey = :location_id

```
render :nothing => true, :status => (update_treeview(modelclass, foreignkey) ? 200 : 500)
```

  end

  private 

  def load_location
    @location = Location.find(params[:id])
  end
end

***\* the generate model is ****
class Location < ActiveRecord::Base
  include BeautifulScaffoldModule      

  before_save :fulltext_field_processing

  def fulltext_field_processing
    # You can preparse with own things here
    generate_fulltext_field([])
  end
  scope :sorting, lambda{ |options|
    attribute = options[:attribute]
    direction = options[:sorting]

```
attribute ||= "id"
direction ||= "DESC"

order("#{attribute} #{direction}")
```

  }
    # You can OVERRIDE this method used in model form and search form (in belongs_to relation)
  def caption
    (self["name"] || self["label"] || self["description"] || "##{id}")
  end
end

*****\* asstes compile warning ***********

BTW please add assets.rb to /config/initializers with

Rails.application.config.assets.precompile += %w( reset.css )
Rails.application.config.assets.precompile += %w( bootstrap.min.css )
Rails.application.config.assets.precompile += %w( bootstrap-responsive.min.css )
Rails.application.config.assets.precompile += %w( datepicker.css )
Rails.application.config.assets.precompile += %w( timepicker.css )
Rails.application.config.assets.precompile += %w( beautiful-scaffold.css )
Rails.application.config.assets.precompile += %w( tagit-dark-grey.css )
Rails.application.config.assets.precompile += %w( colorpicker.css )  
Rails.application.config.assets.precompile += %w( bootstrap-wysihtml5.css )
Rails.application.config.assets.precompile +=%w( jquery.pjax.js )
Rails.application.config.assets.precompile +=%w( jquery.markitup.js )
Rails.application.config.assets.precompile +=%w( markitup/sets/default/set.js )
Rails.application.config.assets.precompile +=%w( markitup/skins/simple/style.css )
Rails.application.config.assets.precompile +=%w( markitup/sets/bbcode/style.css )
Rails.application.config.assets.precompile +=%w( markitup/sets/wiki/style.css)
Rails.application.config.assets.precompile +=%w( markitup/sets/markdown/style.css)
Rails.application.config.assets.precompile +=%w( markitup/sets/html/style.css)
Rails.application.config.assets.precompile +=%w( markitup/sets/textile/style.css)
Rails.application.config.assets.precompile +=%w( markitup/sets/bbcode/set.js)
Rails.application.config.assets.precompile +=%w( markitup/sets/wiki/set.js)
Rails.application.config.assets.precompile +=%w( markitup/sets/markdown/set.js)
Rails.application.config.assets.precompile +=%w( markitup/sets/html/set.js)
Rails.application.config.assets.precompile +=%w( markitup/sets/textile/set.js )
Rails.application.config.assets.precompile +=%w( bootstrap.min.js )
Rails.application.config.assets.precompile +=%w( bootstrap-alert.js )
Rails.application.config.assets.precompile +=%w( bootstrap-datepicker.js )
Rails.application.config.assets.precompile +=%w( bootstrap-timepicker.js )
Rails.application.config.assets.precompile +=%w( bootstrap-dropdown.js )
Rails.application.config.assets.precompile +=%w( bootstrap-modal.js )
Rails.application.config.assets.precompile +=%w( bootstrap-tooltip.js )
Rails.application.config.assets.precompile +=%w( bootstrap-colorpicker.js )
Rails.application.config.assets.precompile +=%w( a-wysihtml5-0.3.0.min.js )
Rails.application.config.assets.precompile +=%w( bootstrap-wysihtml5.js )
Rails.application.config.assets.precompile +=%w( bootstrap-datetimepicker-for-beautiful-scaffold.js )
